client/src/screens/search-groups.screen.js

```diff
@@ -0,0 +1,265 @@
+    fontSize: 11,
+    textAlign: 'right',
+  },
+  groupUsername: {
+    paddingVertical: 4,
+  },
+  header: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+});
+
+// format createdAt with moment
+const formatCreatedAt = createdAt => moment(createdAt).calendar(null, {
+  sameDay: '[Today]',
+  nextDay: '[Tomorrow]',
+  nextWeek: 'dddd',
+  lastDay: '[Yesterday]',
+  lastWeek: 'dddd',
+  sameElse: 'DD/MM/YYYY',
+});
+
+class Group extends Component {
+  constructor(props) {
+    super(props);
+    this.myGroups = this.props.myGroups;
+    this.joinGroupQuery = this.props.joinGroupQuery.bind(this);
+    this.joinGroup = this.props.joinGroup.bind(this, this.props.group);
+
+  }
+
+  render() {
+    const { id, name } = this.props.group;
+    console.log(this.myGroups)
+    const tags = (this.props.group.tags.map((elem) => {return '#'+elem.name})).join(',')
+    //color the already subscribed groups green
+    let iHaveDisAlready = false
+    this.myGroups.some(function(a) {
+      if (a.id == id)
+      {
+        iHaveDisAlready = true;
+        return true;
+      }
+    });
+
+    return (
+      <TouchableHighlight
+        key={id}
+        onPress={this.joinGroup}
+      >
+        <View style={styles.groupContainer}>
+          <Icon name="group" size={24} color={iHaveDisAlready ? 'red' : 'green'} />
+          <View style={styles.groupTextContainer}>
+            <View style={styles.groupTitleContainer}>
+              <Text style={styles.groupName} numberOfLines={1}>{name}</Text>
+              <Text style={styles.groupLastUpdated}>{id}</Text>
+            </View>
+            <Text style={styles.groupText} numberOfLines={1}>{tags}
+            </Text>
+          </View>
+        </View>
+        </TouchableHighlight>
+    );
+  }
+}
+
+Group.propTypes = {
+  joinGroup: PropTypes.func.isRequired,
+  joinGroupQuery: PropTypes.func.isRequired,
+  myGroups: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      name: PropTypes.string.isRequired,
+    }),
+  ),
+  group: PropTypes.shape({
+    id: PropTypes.number,
+    name: PropTypes.string,
+    tags: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        name: PropTypes.string.isRequired,
+      }),
+    ),
+  }),
+};
+
+class AllGroups extends Component {
+  static navigationOptions = {
+    title: 'All Groups',
+    tabBarIcon: ({ tintColor}) => <Icon size={24} name={'group'} color={tintColor} />
+  };
+
+  constructor(props) {
+    super(props);
+    this.onRefresh = this.onRefresh.bind(this);
+  }
+
+  onRefresh() {
+    this.props.refetch();
+  }
+
+  keyExtractor = item => item.id;
+
+  renderItem = ({ item }) => <Group group={item} myGroups={this.props.user.groups} joinGroup={this.joinGroup} joinGroupQuery={this.props.groupUpdate}/>;
+
+  joinGroup(group) {
+    this.props.joinGroupQuery({groupId:group.id}).then((result) => {
+      alert('joined - '+group.name)
```

Comment:
does this even work? We shouldn't need to show confirmation to users but we do need to make sure we display an error if it fails. See login/signup for inspiration.

Reply:
yes it works, long term we want to redraw the whole page to reflect the change but I couldnt work that out at the time and moved on to other more important things.